chore: document sdk helpers (#774)
defenseunicorns/pepr: docs/030_user-guide/030_actions/010_mutate.md

# Mutate

Mutating admission webhooks are invoked first and can modify objects sent to the API server to enforce custom defaults. After an object is sent to Pepr's Mutating Admission Webhook, Pepr will [annotate the object](https://github.com/defenseunicorns/pepr/blob/f01f5eeda16c13ecd0d51b26b8a16ed7e4c1b080/src/lib/mutate-processor.ts#L64) to indicate the status.

After a successful mutation of an object in a module with UUID static-test, and capability name hello-pepr, expect to see this annotation: `static-test.pepr.dev/hello-pepr: succeeded`.

> See also [SDK](../130_sdk.md) for information on the Pepr SDK.

Reviewer: ```suggestion
```
Feels like this it'd be better placed at the end of the Mutate Helpers section -- under a "See Also" header, maybe?

Author: Moved to the end.

## Mutate Helpers

### SetLabel

### SetAnnotation

### RemoveAnnotation

> ### RemoveLabel

Reviewer: ```suggestion
### RemoveLabel

### See Also

The Pepr Dev [SDK](../130_sdk.md).
```

  });
```

### `SetAnnotation`

`SetAnnotation` is used to set an annotation on a Kubernetes object as part of a Pepr Mutate action.

For example, to add an annotation when a ConfigMap is created:

```typescript
When(a.ConfigMap)
  .IsCreated()
  .Mutate(request => {
    request
      // Here we are adding an annotation to the ConfigMap.
      .SetAnnotation("pepr.dev", "annotations-work-too");

    // Note that we are not returning anything here. This is because Pepr is tracking the changes in each action automatically.
  });
```

### `RemoveAnnotation`

`RemoveAnnotation` is used to remove an annotation on a Kubernetes object as part of a Pepr Mutate action.

For example, to remove an annotation when a ConfigMap is updated:

```typescript
When(a.ConfigMap)
  .IsUpdated()
  .Mutate(request => {
    request
      // Here we are removing an annotation from the ConfigMap.
      .RemoveAnnotation("remove-me");

    // Note that we are not returning anything here. This is because Pepr is tracking the changes in each action automatically.
  });
```

## See Also

> See also [SDK](../130_sdk.md) for information on the Pepr SDK.

Reviewer: ```suggestion
Looking for some more generic helpers?  Check out the [Module Author SDK](../130_sdk.md) for information on other things that Pepr can help with.
```
Not 100% on the wording, but something like this would set reader expectation a bit better cuz it set context on the SDK as being "more generic helpers".  Kinda hard to tell what "the Pepr SDK" is / how it would be relevant, otherwise.
